src/sirc/sirc_event_hdr.c

```diff
@@ -72,11 +72,15 @@ void sirc_event_hdr(SircSession *sirc, SircMessage *imsg){
             case SIRC_RFC_RPL_WELCOME:
                 g_return_if_fail(events->welcome);
                 events->welcome(sirc, num, origin, params, imsg->nparam);
-                /* Do not break here */
-            default:
-                g_return_if_fail(events->numeric);
-                events->numeric(sirc, num, origin, params, imsg->nparam);
+                break;
+            case SIRC_RFC_RPL_UMODEIS:
+                 /* User mode changed */
+                 g_return_if_fail(events->umode);
+                 events->umode(sirc, imsg->cmd, origin, params, imsg->nparam);
+                 break;
         }
+        g_return_if_fail(events->numeric);
+        events->numeric(sirc, num, origin, params, imsg->nparam);
```

Comment:
`event->numeric` will be called after `break` out from every case, I think it is unexpected.

Reply:
I thought it could be confusing if `irc_event_numeric` was called for all numerics except a few

Reply:
It is a fallback of "non-special" numeric event, it maybe a little confusing, but call it for all numerics is incorrect for now.

Reply:
You should not move `event->numeric` out of default case of switch clause.